src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -372,10 +372,38 @@ private NamespaceSymbol ResolveExternAliasTarget(BindingDiagnosticBag diagnostic
             return target;
         }
 
-        private NamespaceOrTypeSymbol ResolveAliasTarget(NameSyntax syntax, BindingDiagnosticBag diagnostics, ConsList<TypeSymbol>? basesBeingResolved)
+        private NamespaceOrTypeSymbol ResolveAliasTarget(
+            UsingDirectiveSyntax usingDirective,
+            BindingDiagnosticBag diagnostics,
+            ConsList<TypeSymbol>? basesBeingResolved)
         {
-            var declarationBinder = ContainingSymbol.DeclaringCompilation.GetBinderFactory(syntax.SyntaxTree).GetBinder(syntax).WithAdditionalFlags(BinderFlags.SuppressConstraintChecks | BinderFlags.SuppressObsoleteChecks);
-            return declarationBinder.BindNamespaceOrTypeSymbol(syntax, diagnostics, basesBeingResolved).NamespaceOrTypeSymbol;
+            if (usingDirective.Type is not NameSyntax)
+                MessageID.IDS_FeatureUsingTypeAlias.CheckFeatureAvailability(diagnostics, usingDirective.Type);
+
+            var syntax = usingDirective.Type;
+            var flags = BinderFlags.SuppressConstraintChecks | BinderFlags.SuppressObsoleteChecks;
+            if (usingDirective.UnsafeKeyword != default)
```

Comment:
We should also check that the compilation has the unsafe mode enabled (see `AllowUnsafe` in compilation options).

Reply:
Done.  I added teh test, and the behavior is as you would expect.  With the option off, you get an error as long as you use `using unsafe` (i.e. you don't need to actually use a pointer type in the alias to get teh error).